### PR TITLE
 Make UNIX socket configurable 

### PR DESCRIPTION
**Pull request by Bond-009**

### CODE REVIEW

**Comment by joshuaboniface** [comment]
What is the socket even used for? I've never noticed it before. 100% it should be configurable though.

**Comment by crobibero** [comment]
> What is the socket even used for? I've never noticed it before. 100% it should be configurable though.

It was added in https://github.com/jellyfin/jellyfin/pull/3532

File: MediaBrowser.Controller/Extensions/ConfigurationExtensions.cs [hunk 1]
```diff
@@ -33,6 +33,16 @@ public static class ConfigurationExtensions
...
 
+        /// <summary>
+        /// The key for a setting that indicates whether kestrel should bind to a unix socket.
+        /// </summary>
+        public const string BindToUnixSocketKey = "kerstrel:socket";
+
+        /// <summary>
+        /// The key for the unix socket path.
+        /// </summary>
+        public const string UnixSocketPathKey = "kerstrel:socketPath";
```
Comment (DMouse10462) [inline MediaBrowser.Controller/Extensions/ConfigurationExtensions.cs:44]:
```suggestion
        public const string BindToUnixSocketKey = "kestrel:socket";

        /// <summary>
        /// The key for the unix socket path.
        /// </summary>
        public const string UnixSocketPathKey = "kestrel:socketPath";
```

File: MediaBrowser.Controller/Extensions/ConfigurationExtensions.cs [hunk 2]
```diff
@@ -65,5 +75,21 @@ public static string GetFFmpegAnalyzeDuration(this IConfiguration configuration)
...
+        public static bool UseUnixSocket(this IConfiguration configuration)
+            => configuration.GetValue<bool>(BindToUnixSocketKey);
+
+        /// <summary>
+        /// Gets the path for the unix socket from the <see cref="IConfiguration" />.
+        /// </summary>
+        /// <param name="configuration">The configuration to read the setting from.</param>
+        /// <returns>The unix socket path.</returns>
+        public static string GetUnixSocketPath(this IConfiguration configuration)
+            => configuration[FfmpegAnalyzeDurationKey];
```
Comment (Ullmie02) [inline MediaBrowser.Controller/Extensions/ConfigurationExtensions.cs:93]:
```suggestion
            => configuration[UnixSocketPathKey];
```

**Comment by g2p** [comment]
Not sure how to add to the documentation, but here is how to make use of this:

```
cat /etc/systemd/system/jellyfin.service.d/jellyfin.service.conf
[Service]
RuntimeDirectory=jellyfin
Environment=JELLYFIN_kestrel__socket=true
Environment=JELLYFIN_kestrel__socketPath=/run/jellyfin/socket
```

**Comment by cyberb** [comment]
is it possible to disable tcp and just listen on a unix socket?

**Pull request merged**